kamyroll_gui/utils/api.py

```diff
@@ -112,29 +113,33 @@ def call_api(path, /, params=None):
     return json_data
 
 
+error_codes = Enum("premium_only", "bad_player_connection", "bad_initialize")
+
 def _handle_error(code, message, use_login, channel_id):
     _logger.error("Api returned error code %s: %s", code, message)
-    if code == "premium_only":
-        if use_login:
-            message += "\nConsider using the premium bypass"
-            raise ApiError(message)
 
-        if channel_id == "funimation":
-            return False
-        # this should only ever happen if
-        # the api backend user runs out of premium
-        raise ApiError("Unexpected bypass error, try again later")
+    match code:
+        case "premium_only":
+            if use_login:
+                message += "\nConsider using the premium bypass"
+                raise ApiError(message)
+
+            if channel_id == "funimation":
+                return False
+            # this should only ever happen if
+            # the api backend user runs out of premium
+            raise ApiError("Unexpected bypass error, try again later")
 
-    if code == "bad_player_connection":
-        wait(2000)
-        return
+        case "bad_player_connection":
+            wait(2000)
+            return
 
-    if code == "bad_initialize":
-        wait(2000)
-        return
+        case "bad_initialize":
+            wait(2000)
+            return
 
-    if code in ["unknown_id"]:
-        raise ApiError("The provided id of the url is not valid")
+        case _:
+            raise ApiError(f"Error: {code}. The provided id of the url is not valid")
```

Comment:
I see no point in using the match statement here. It additionally changes behaviour to always raise an `ApiError` instead of retrying

Reply:
Match is a nice tool that can be used here. It's an opportunity to use a new feature ;) The newer commit resolves the issues mentioned here.